src/log/log.c

```diff
@@ -1403,7 +1403,7 @@ __log_truncate_file(WT_SESSION_IMPL *session, WT_FH *log_fh, wt_off_t offset)
     conn = S2C(session);
     log = conn->log;
 
-    if (!F_ISSET(log, WT_LOG_TRUNCATE_NOTSUP) && conn->hot_backup_start == 0) {
+    if (log != NULL && !F_ISSET(log, WT_LOG_TRUNCATE_NOTSUP) && conn->hot_backup_start == 0) {
```

Comment:
@sueloverso, since the `conn->log` variable can be `NULL`, I wonder if that check is too late. The stack trace from [WT-12403](https://jira.mongodb.org/browse/WT-12403) shows we are failing in `__log_truncate_file` but I wonder if we should already exit earlier when we detect `conn->log` is `NULL`. What do you think?